src/constants/api.ts

```diff
@@ -119,30 +138,34 @@ export const getLeavesBySquad = async (squad: Squad) => {
             const absenceDays = leavesData.data.items.length / 2;
             const presenceDays = totalDays - absenceDays;
 
+            const groupedByDate: GroupedByDate = _.groupBy(leavesData.data.items, 'date');
+
             const user = (await allTechUsers()).items.find(user => user.id === userId);
             if (user) {
                 console.log(`${user.firstName} ${user.lastName} sera présent(e) ${presenceDays} jours sur 10`);
             }
-            if (user && leavesData.data.items.length > 0) {
+            if (user && Object.keys(groupedByDate).length > 0) {
                 console.log(`Jours d'absence pour ${user.firstName} ${user.lastName}:`);
-                const allDay: string[] = [];
-                leavesData.data.items.forEach(item => {
-                    allDay.push(item.date);
-                    
-                    const am = item.isAM;
-                });
-                    const uniqueDay = allDay.filter((obj, index, self) => self.findIndex(o => JSON.stringify(o) === JSON.stringify(obj)) === index);
-                    uniqueDay.forEach(day =>{
-                        console.log((dayjs(day).format(`DD/MM/YYYY`)));
-
-                    })
+
+                for (const date in groupedByDate) {
+                    const items: LeaveItem[] = groupedByDate[date];
+                    const amLeave: LeaveItem = items.find(item => item.isAM === true)!;
+                    const pmLeave: LeaveItem = items.find(item => item.isAM === false)!;
+
+                    if (amLeave && pmLeave) {
+                        console.log(dayjs(date).format('DD/MM/YYYY'));
+                    } else if (amLeave) {
+                        console.log(`${dayjs(date).format('DD/MM/YYYY')} - Matin`);
+                    } else if (pmLeave) {
+                        console.log(`${dayjs(date).format('DD/MM/YYYY')} - Après-midi`);
+                    }
```

Comment:
Pas besoin de else ou else if ici, car toujours valide quoi qu'il arrive :
```suggestion
                    console.log(`${dayjs(date).format('DD/MM/YYYY')} - ${amLeave ? 'Matin' : 'Après-midi'}`);
```